sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveExternalCatalog.scala

```diff
@@ -383,21 +383,36 @@ private[spark] class HiveExternalCatalog(conf: SparkConf, hadoopConf: Configurat
     requireTableExists(db, tableDefinition.identifier.table)
     verifyTableProperties(tableDefinition)
 
-    if (DDLUtils.isDatasourceTable(tableDefinition)) {
-      val oldDef = client.getTable(db, tableDefinition.identifier.table)
+    // convert table statistics to properties so that we can persist them through hive api
+    val catalogTable = if (tableDefinition.catalogStats.isDefined) {
+      // first we need to clear the old (i.e. inaccurate) stats
```

Comment:
yeah, it's better to move the logic from **HiveClientImpl** into **restoreTableMeta**